includes/gateway/abstract-omise-payment-offline.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+defined( 'ABSPATH' ) || exit;
+
+require_once dirname( __FILE__ ) . '/class-omise-payment.php';
+
+/**
+ * @since 4.0
+ */
+abstract class Omise_Payment_Offline extends Omise_Payment {
+	/**
+	 * A string of Omise Source's type
+	 * (expecting for either 'paynow' or 'bill_payment_tesco_lotus').
+	 *
+	 * @var string
+	 */
+	protected $source_type;
+
+	/**
+	 * @inheritdoc
+	 */
+	public function charge( $order_id, $order ) {
+		$total    = $order->get_total();
+		$currency = $order->get_order_currency();
+		$metadata = array_merge(
+			apply_filters( 'omise_charge_params_metadata', array(), $order ),
+			array( 'order_id' => $order_id ) // override order_id as a reference for webhook handlers.
+		);
+
+		return OmiseCharge::create( array(
+			'amount'      => Omise_Money::to_subunit( $total, $currency ),
+			'currency'    => $currency,
+			'description' => apply_filters( 'omise_charge_params_description', 'WooCommerce Order id ' . $order_id, $order ),
+			'source'      => array( 'type' => $this->source_type ),
+			'metadata'    => $metadata
+		) );
```

Comment:
Note that `return_uri` has been removed. It is because Offline payment methods don't require for the `return_uri` parameter.